### PR TITLE
feat: remove orgId dependency in authz

### PR DESCRIPTION
**Pull request by niharbansal02**

### CODE REVIEW

**Comment by AkarshSatija** [comment]
Can we mention all the APIs being affected by this?

File: internal/proxy/hook/authz/authz.go [hunk 1]
```diff
@@ -30,12 +32,18 @@ type Authz struct {
 	// To skip all the next hooks and just respond back
 	escape hook.Service
 
+	Deps api.Deps
```
Comment (mabdh) [inline internal/proxy/hook/authz/authz.go:35]: Don't add deps dependency, use project service interface that you have defined instead

File: internal/proxy/hook/authz/authz.go [hunk 2]
```diff
@@ -194,16 +202,26 @@ func (a Authz) ServeHook(res *http.Response, err error) (*http.Response, error)
 	return a.next.ServeHook(res, nil)
 }
 
-func createResources(permissionAttributes map[string]interface{}) ([]resource.Resource, error) {
+func createResources(ctx context.Context, permissionAttributes map[string]interface{}, p ProjectService) ([]resource.Resource, error) {
 	var resources []resource.Resource
 	projects, err := getAttributesValues(permissionAttributes["project"])
 	if err != nil {
 		return nil, err
 	}
```
Comment (mabdh) [inline internal/proxy/hook/authz/authz.go:211]: Is it really needed to pass the project service as argument? If not, better to do this
```suggestion
func (a Authz) createResources(ctx context.Context, permissionAttributes map[string]interface{}) ([]resource.Resource, error) {
	var resources []resource.Resource
	projects, err := getAttributesValues(permissionAttributes["project"])
	if err != nil {
		return nil, err
	}
```

File: internal/proxy/hook/authz/authz.go [hunk 3]
```diff
@@ -194,16 +202,26 @@ func (a Authz) ServeHook(res *http.Response, err error) (*http.Response, error)
...
 
-	orgs, err := getAttributesValues(permissionAttributes["organization"])
-	if err != nil {
-		return nil, err
+	var organizationIds []string
+	var projectIds []string
+	for _, proj := range projects {
+		project, err := p.Get(ctx, proj)
```
Comment (mabdh) [inline internal/proxy/hook/authz/authz.go:215]: With the previous changes, here you could do
```suggestion
		project, err := a.projectService.Get(ctx, proj)
```

File: internal/proxy/hook/authz/authz.go [hunk 4]
```diff
@@ -177,7 +185,7 @@ func (a Authz) ServeHook(res *http.Response, err error) (*http.Response, error)
 		attributes[key] = value
 	}
 
-	resources, err := createResources(attributes)
+	resources, err := createResources(res.Request.Context(), attributes, a.projectService)
```
Comment (mabdh) [inline internal/proxy/hook/authz/authz.go:188]: With the changes from line 205, you could call `createResources`
```suggestion
	resources, err := a.createResources(res.Request.Context(), attributes)
```

**Review by mabdh — CHANGES_REQUESTED** [review]

**Review by AkarshSatija — APPROVED** [review]

File: internal/proxy/hook/authz/authz.go (hunk 1 above)
Comment (niharbansal02) [reply internal/proxy/hook/authz/authz.go]: Used Project Service to Get Organization.

**Pull request merged**